moveit_core/collision_detection/include/moveit/collision_detection/world.h

```diff
@@ -44,12 +44,15 @@
 #include <map>
 #include <memory>
 #include <boost/function.hpp>
+#include <boost/shared_ptr.hpp>
 #include <Eigen/Geometry>
 #include <eigen_stl_containers/eigen_stl_vector_container.h>
 
 namespace shapes
 {
-MOVEIT_CLASS_FORWARD(Shape);
+class Shape;
+typedef boost::shared_ptr<Shape> ShapePtr;
+typedef boost::shared_ptr<const Shape> ShapeConstPtr;
```

Comment:
I agree, it would be nice to have `std::shared_ptr` there. The previous deviation from REP-003 was out of necessity though. On the other hand, `urdfdom` also switched. And while it may not be real  ROS package that is more or less a technicality.

But yeah, as long as the `geometric_shape` API uses `boost::shared_ptr` we can't switch them here.